bot/exts/utilities/color.py

```diff
@@ -0,0 +1,337 @@
+"""
+
+COLOR_JSON_PATH = "bot/resources/utilities/ryanzec_colours.json"
+with open(COLOR_JSON_PATH) as f:
+    COLOR_MAPPING = json.load(f)
+
+
+class Color(commands.Cog):
+    """User initiated commands to receive color information."""
+
+    def __init__(self, bot: Bot):
+        self.bot = bot
+
+    @commands.command(aliases=["colour"])
+    async def color(self, ctx: commands.Context, mode: str, *, user_color: str) -> None:
+        """
+        Send information on input color code or color name.
+
+        Possible modes are: "hex", "rgb", "hsv", "hsl", "cmyk" or "name".
+        """
+        logger.debug(f"{mode = }")
+        logger.debug(f"{user_color = }")
+        if mode.lower() == "hex":
+            await self.hex_to_rgb(ctx, user_color)
+        elif mode.lower() == "rgb":
+            rgb_color = self.tuple_create(user_color)
+            await self.color_embed(ctx, rgb_color)
+        elif mode.lower() == "hsv":
+            await self.hsv_to_rgb(ctx, user_color)
+        elif mode.lower() == "hsl":
+            await self.hsl_to_rgb(ctx, user_color)
+        elif mode.lower() == "cmyk":
+            await self.cmyk_to_rgb(ctx, user_color)
+        elif mode.lower() == "name":
+            color_name, hex_color = self.match_color_name(user_color)
+            if "#" in hex_color:
+                rgb_color = ImageColor.getcolor(hex_color, "RGB")
+            else:
+                rgb_color = ImageColor.getcolor("#" + hex_color, "RGB")
+            await self.color_embed(ctx, rgb_color, color_name)
+        else:
+            # mode is either None or an invalid code
+            if mode is None:
+                no_mode_embed = Embed(
+                    title="No 'mode' was passed, please define a color code.",
+                    description="Possible modes are: Name, Hex, RGB, HSV, HSL and CMYK.",
+                    color=Colours.soft_red,
+                )
+                await ctx.send(embed=no_mode_embed)
+                return
+            wrong_mode_embed = Embed(
+                title=f"The color code {mode} is not a valid option",
+                description="Possible modes are: Name, Hex, RGB, HSV, HSL and CMYK.",
+                color=Colours.soft_red,
+            )
+            await ctx.send(embed=wrong_mode_embed)
+            return
+
+    @staticmethod
+    def tuple_create(input_color: str) -> tuple[int, int, int]:
+        """
+        Create a tuple of integers based on user's input.
+
+        Can handle inputs of the types:
+        (100, 100, 100)
+        100, 100, 100
+        100 100 100
+        """
+        if "(" in input_color:
+            remove = "[() ]"
+            color_tuple = re.sub(remove, "", input_color)
+            color_tuple = tuple(map(int, color_tuple.split(",")))
+        elif "," in input_color:
+            color_tuple = tuple(map(int, input_color.split(",")))
+        else:
+            color_tuple = tuple(map(int, input_color.split(" ")))
+        return color_tuple
```

Comment:
What happens if the conversion fails?

Reply:
Ahh yes I see what you mean. I'll add an error embed message if it cannot be split.

Reply:
Yeah. Although possibly this function should raise an exception, which can then be caught by the caller and an embed sent.

I sort of don't like functions like this having side effects. I usually aim to write pure functions if possible, which are then tied together with imperative code.

Reply:
I'm not sure the last part of my above comment made sense 😄 Ask if you don't get what I mean.